src/main/java/htsjdk/samtools/SAMTag.java

```diff
@@ -105,24 +107,46 @@
     U2,
     UQ;
 
-    private final short shortValue = SAMTag.makeBinaryTag(this.name());;
+    // Cache of already-converted tags.  Should speed up SAM text generation.
+    // Not synchronized because race condition is not a problem.
+    private static final String[] stringTags = new String[Short.MAX_VALUE];
```

Comment:
static fields should be uppercase
```suggestion
    private static final String[] STRING_TAGS = new String[Short.MAX_VALUE];
```

Reply:
This isn't really a constant though.  My thought is that things that are constant are final, but this is a static cache that's updated with new values.  It seems more confusing to make it caps since people assume those are immutable.